src/enums/index.ts

```diff
@@ -1,138 +1,153 @@
-import { makeClass } from "./../utils/make-class";
-const assert = require("assert");
-const _ = require("lodash");
-const { parseBytes, serializeUIntN } = require("./../utils/bytes-utils");
-const enums = require("./definitions.json");
-
-function transformWith(func, obj) {
-  return _.transform(obj, func);
+import { serializeUIntN } from "./../utils/bytes-utils";
+import * as enums from "./definitions.json";
+
+const TYPE_WIDTH = 2;
+const LEDGER_ENTRY_WIDTH = 2;
+const TXN_TYPE_WIDTH = 2;
+const TXN_RESULT_WIDTH = 1;
+
+class EnumType {
+  bytes: Uint8Array;
+
+  constructor(
+    public name: string,
+    public ordinal: number,
+    public ordinalWidth: number
+  ) {
+    this.bytes = serializeUIntN(ordinal, ordinalWidth);
+  }
+
+  toJSON(): string {
+    return this.name;
+  }
+
+  toBytesSink(sink): void {
+    sink.put(this.bytes);
+  }
+}
+
+class EnumTypes {
+  constructor(types: { [key: string]: number }, public ordinalWidth: number) {
+    Object.entries(types).forEach(([k, v]) => {
+      this[k] = new EnumType(k, v, ordinalWidth);
+      this[v.toString()] = this[k];
+    });
+  }
+
+  from(value: EnumType | string): EnumType {
+    return value instanceof EnumType ? value : (this[value] as EnumType);
+  }
+
+  fromParser(parser): EnumType {
+    return this.from(parser.readUIntN(this.ordinalWidth).toString());
+  }
 }
 
-function biMap(obj, valueKey) {
-  return _.transform(obj, (result, value, key) => {
-    result[key] = value;
-    result[value[valueKey]] = value;
-  });
+type FieldInfo = {
+  nth: number;
+  isVLEncoded: boolean;
+  isSerialized: boolean;
+  isSigningField: boolean;
+  type: string;
+};
+
+class Field {
+  nth: number;
+  isVLEncoded: boolean;
+  isSerialized: boolean;
+  isSigningField: boolean;
+  type: EnumType;
+  ordinal: number;
+  name: string;
+  bytes: Uint8Array;
+  associatedType: any; // Will change this to type CoreType when CoreType is refactored
```

Comment:
Similar here, associatedType is of type SerializedType, but I have yet to rewrite src/types, so it will have to be changed later